### ISSUE TITLE
Programs: Is it in the goals building and running other C programs directly from Gramado?

### ISSUE THREAD

**Issue by takusuman** [frednora/gramado#42]
G'day to Gramado developers.
I've been following this work since it started (I think since 2020, if I'm not mistaken) and I was thinking if it would be possible to build other C programs written to UNIX on it.
I haven't looked at the source code and documentation yet, but I'm asking this because I thought about building the Heirloom NG set of utilities on it and see if it can run/the toolset that I maintain is portable enough.

**Comment by frednora** [comment]
> 

I have some few programs using only the libc. Take a look at the folder gramado/core/commands/  ... you're gonna see some programs using only the libc ... the problem is that this kind of application is printing in a raw mode ... not inside the client area of a terminal window yet, but  in the fullscreen ... those are ring3 programs ... stdout is a tty ... when you send data to stdout you're gonna print out in the foreground kernel console.

**Comment by frednora** [comment]
Yes. it is. The plain is create unix-like commands using the libc and client-side gui apllications using the libgws. Take a look at  gramado/core/commands/  and gramado/desktop/apps/. :)

**Comment by takusuman** [comment]
> > 
> 
> I have some few programs using only the libc. Take a look at the folder gramado/core/commands/ ... you're gonna see some programs using only the libc ... the problem is that this kind of application is printing in a raw mode ... not inside the client area of a terminal window yet, but in the fullscreen ... those are ring3 programs ... stdout is a tty ... when you send data to stdout you're gonna print out in the foreground kernel console.

I've took a look at some of the commands and saw a little about the C library. I'd need to study Gramado more before talking anything about the C library, by the way.

I would propose that Gramado had some sort of compatibility with POSIX, but I don't know if "being an UNIX-compatible" is part of the goals of Gramado, so maybe it could be just a "compatibility layer".
